qiskit_experiments/library/randomized_benchmarking/clifford_utils.py

```diff
@@ -375,328 +294,322 @@ def _unpack_num(num, sig):
+    Number of Cliffords == 36."""
+    circuits = []
+    num_h = [0, 1]
+    v_w_gates = ["i", "v", "w"]
+    for h0, h1, v0, v1 in itertools.product(num_h, num_h, v_w_gates, v_w_gates):
+        qc = QuantumCircuit(2)
+        for _ in range(h0):
+            qc.h(0)
+        for _ in range(h1):
+            qc.h(1)
+        _append_v_w(qc, v0, v1)
+        circuits.append(qc)
+    return circuits
+
+
+def _create_cliff_2q_layer_1():
+    """Layer 1 consists of one of the following:
+    - nothing
+    - cx(0,1) followed by 0/1/2 V gates on each qubit
+    - cx(0,1), cx(1,0) followed by 0/1/2 V gates on each qubit
+    - cx(0,1), cx(1,0), cx(0,1)
+    Number of Cliffords == 20."""
+    circuits = [QuantumCircuit(2)]  # identity at the beginning
+
+    v_w_gates = ["i", "v", "w"]
+    for v0, v1 in itertools.product(v_w_gates, v_w_gates):
+        qc = QuantumCircuit(2)
         qc.cx(0, 1)
-        transpiled = transpile(
-            qc, optimization_level=1, basis_gates=self.basis_gates, backend=self._backend
-        )
-        self._transpiled_cliff_layer[1].append(transpiled)
-
-    def _transpile_cliff_layer_2(self):
-        """Layer 2 consists of a Pauli gate on each qubit {Id, X, Y, Z}.
-        Number of Cliffords == 16."""
-        if self._transpiled_cliff_layer[2] != []:
-            return
-
-        pauli = ("i", XGate(), YGate(), ZGate())
-        for p0, p1 in itertools.product(pauli, pauli):
-            qc = QuantumCircuit(2)
-            if p0 != "i":
-                qc.append(p0, [0])
-            if p1 != "i":
-                qc.append(p1, [1])
-
-            transpiled = transpile(
-                qc, optimization_level=1, basis_gates=self.basis_gates, backend=self._backend
-            )
-            self._transpiled_cliff_layer[2].append(transpiled)
+        _append_v_w(qc, v0, v1)
+        circuits.append(qc)
 
-    def create_random_clifford(self, rng: Generator) -> QuantumCircuit:
-        """For 1-qubit, select a random transpiled Clifford.
-
-         Returns:
-            A random Clifford represented as a :class:`QuantumCircuit`.
+    for v0, v1 in itertools.product(v_w_gates, v_w_gates):
+        qc = QuantumCircuit(2)
+        qc.cx(0, 1)
+        qc.cx(1, 0)
+        _append_v_w(qc, v0, v1)
+        circuits.append(qc)
+
+    qc = QuantumCircuit(2)  # swap at the end
+    qc.cx(0, 1)
+    qc.cx(1, 0)
+    qc.cx(0, 1)
+    circuits.append(qc)
+    return circuits
+
+
+def _create_cliff_2q_layer_2():
+    """Layer 2 consists of a Pauli gate on each qubit {Id, X, Y, Z}.
+    Number of Cliffords == 16."""
+    circuits = []
+    pauli = ("i", XGate(), YGate(), ZGate())
+    for p0, p1 in itertools.product(pauli, pauli):
+        qc = QuantumCircuit(2)
+        if p0 != "i":
+            qc.append(p0, [0])
+        if p1 != "i":
+            qc.append(p1, [1])
+        circuits.append(qc)
+    return circuits
+
+
+_CLIFFORD_LAYER = (
```

Comment:
What is the expected overhead of this at a time of importing this module?

Reply:
It's almost invisible. Previously importing clifford_utils takes about 1.41--1.43 seconds and now it takes about 1.44--1.46 seconds. About 0.03 sec increase seems acceptable to me.